crowbar_framework/app/models/node_object.rb

```diff
@@ -528,18 +528,23 @@ def state
     return "unknown" if (@node.nil? or @role.nil?)
     if self.crowbar["state"] === "ready" and @node["ohai_time"]
       since_last = Time.now.to_i-@node["ohai_time"].to_i
-      crowbar_proposal = Proposal.where(barclamp: "provisioner").first
-      max_last = if crowbar_proposal
-        # time from proposal + 5 min buffer time
-        crowbar_proposal.properties["attributes"]["provisioner"]["chef_client_runs"] + 300
-      else
-        1200 # 20min
-      end
       return "noupdate" if since_last > max_last
     end
     return self.crowbar["state"] || "unknown"
   end
 
+  def max_last
+    frequent_time = Rails.cache.read("chef_client_runs")
+    return frequent_time + 300 unless frequent_time.blank?
+
+    proposal = Proposal.where(barclamp: "provisioner").first
```

Comment:
As I asked previously, why you are querying the proposal this way? Maybe you already answered by email and I didn't see, since I'm way behind on email currently.  But this is inconsistent with the way pretty much everywhere else we read attributes from proposals for a specific node.  The proposal's attributes are already applied to the node via the proposal role, so you can simply read the attribute directly off the Chef node object.